source/common/upstream/health_checker_impl.h

```diff
@@ -417,5 +421,102 @@ class RedisHealthCheckerImpl : public HealthCheckerImplBase {
   Redis::ConnPool::ClientFactory& client_factory_;
 };
 
+/**
+ * gRPC health checker implementation.
+ */
+class GrpcHealthCheckerImpl : public HealthCheckerImplBase {
+public:
+  GrpcHealthCheckerImpl(const Cluster& cluster, const envoy::api::v2::HealthCheck& config,
+                        Event::Dispatcher& dispatcher, Runtime::Loader& runtime,
+                        Runtime::RandomGenerator& random);
+
+private:
+  struct GrpcActiveHealthCheckSession : public ActiveHealthCheckSession,
+                                        public Http::StreamDecoder,
+                                        public Http::StreamCallbacks {
+    GrpcActiveHealthCheckSession(GrpcHealthCheckerImpl& parent, HostSharedPtr host);
```

Comment:
nit: `const HostSharedPtr& host` (Please fix elsewhere in this file in old code if similar).

Reply:
done